src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -745,18 +797,42 @@ await ExecuteWithFailOverPolicyAsync(
 
                 Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
                 SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+
+                List<KeyValueWatcher> selectedKeyValueWatchers = new List<KeyValueWatcher>();
+
+                if (_options.RefreshOptions.RegisterAllEnabled)
+                {
+                    foreach (KeyValueIdentifier keyValueIdentifier in watchedSelectedKeyValueCollections.Keys)
+                    {
+                        selectedKeyValueWatchers.AppendUnique(new KeyValueWatcher()
```

Comment:
There won't be duplicated selectors in _options.KeyValueSelctors, since we have deduped [here](https://github.com/Azure/AppConfiguration-DotnetProvider/blob/main/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs#L201). So, watchedSelectedKeyValueCollections won't have duplicated keys either.
In this case, `AppendUnique` is unnecessary. Did I miss anything?

Reply:
That seems right, I don't see why I added this at the moment.